### PR TITLE
256 - veracode scans and security fixes

### LINKED ISSUE #256: SOHO-7066: Veracode - Fix Vulnerabilities.
**Background**

The [Infor ISO team](https://wiki.infor.com/confluence/display/ISO/Security+Scanning+FAQ) requires that we scan using the Veracode Tool.

This ticket was added to handle this task.

- [X] Assess Initial Scans and Get Access to the Veracode Tool
- [ ] Address errors one by one (on src folder and sample app and Angular)
- [ ] Add mitigation notes for errors
- [ ] Add code tests
- [x] Look into veracode testing tool
- [ ] Document any white listed tags on each component page.

**Notable Fixes**

Insecure HTTP Communication
1.  Search and replaced and removed all xmlns:xlink="http://www.w3.org/1999/xlink" from inline svg (use elements) (4)
1.

### LINKED ISSUE #257: SOHO-7019: XSS in Editor (Homepages Notes widget)
When using the Notes widget in Homepages (which uses the Editor SoHo component), you can insert script tags in hyperlinks:
!xss_notes.gif|thumbnail!

This is pretty dangerous, since it makes it trivial to perform an XSS attack on any platform where users can share the contents of the Editor (assuming no escaping is done when serializing/transmitting the data).

I guess this affects any product that uses the Editor, and is not specific to Homepages widgets.

*Stpes to reproduce*
- Go to http://usalvlhlpool1.infor.com/4.4.0-rc/components/editor/example-clickable-links
- Click the insert anchor button.
- Add a link with URL: "><script>alert()</script>
- Click Insert
- Click the HTML button
_issue occurs here_

Expected: No alert
Actual: Alert appears due to injected script being executed.

### PR DESCRIPTION
**Pull request by tmcconechy**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

As part of ISO (Infor Security Office), we need to satisfy veracode scanner output. This PR adds xssUtils and DOM utils that solve this. Areas the scanner report have been updated.

Also, Added more error checking tests for more component and tests in general. 
And solved several open security issues.
And started in on removing all inline styles #628

**Related github/jira issue (required)**:
Closes #256 
Closes #257

**Steps necessary to review your pull request (required)**:
- Review test output
- Run kitchen sink page to test major example-index pages
- Smoke test major areas

**Additional Info**:
@EdwardCoyle Please go into Veracode and approve/reject the mitigations.

### CODE REVIEW

**Comment by davidcarlsonberg** [comment]
Hey @tmcconechy , I accidentally hit approve while this is obviously failing on a couple tests. I meant to "request changes". Just FYI.

File: app/views/components/field-options/example-index.html [hunk 1]
```diff
@@ -176,14 +176,14 @@
       <div class="field-options">
         <input type="radio" class="radio" name="options" id="option1" value="option1" />
         <label for="option1" class="radio-label">Option one</label>
-        <br>
-        <input type="radio" class="radio" name="options" id="option2" value="option1" checked />
+        <br/>
+        <input type="radio" class="radio" name="options" id="option2" value="option1" checked="true" />
         <label for="option2"  class="radio-label">Option two</label>
-        <br>
+        <br//>
```
Comment (clepore) [inline app/views/components/field-options/example-index.html]: Typo

**Comment by tmcconechy** [comment]
@davidcarlsonberg All the tests seem to pass on the branch checks. Are you sure you have the server running on another terminal session (`npm run start`). https://github.com/infor-design/enterprise/pull/635/checks?check_run_id=13248314 I also reran the tests locally again and they all work for me. 

@clepore I fixed the typo.
In retrospect i could have done this in a few PR's so i agree there. For next time...

**Comment by EdwardCoyle** [comment]
@tmcconechy regarding this:

> Please go into Veracode and approve/reject the mitigations.

Which scans from which date(s) do I need to approve?  I see a ton of them upon logging in

**Comment by tmcconechy** [comment]
Noting that @EdwardCoyle and i chatted on the mitigation results and this does not depend on this PR. So this should be good to go. We will work on that separately / additionally.

**Pull request merged**